fscollections/views.py

```diff
@@ -0,0 +1,226 @@
+
+        if form.is_valid():
+            saved_collection = form.save()
+            # TODO: moderation of CollectionSounds to be accounted for users who are neither maintainers nor owners
+            CollectionSound(user=request.user, collection=saved_collection, sound=sound, status="OK").save()
+            saved_collection.num_sounds =+ 1 #this should be done with a signal/method in Collection models
+            saved_collection.save()
+            msg_to_return = f'Sound "{sound.original_filename}" saved under collection {saved_collection.name}'
+            return JsonResponse('message', msg_to_return)
+        else:
+            msg_to_return = 'This sound already exists in this category'
+            return JsonResponse('message', msg_to_return)
+        
+
+def delete_sound_from_collection(request, collectionsound_id):
+    #this should work as in Packs - select several sounds and remove them all at once from the collection
+    #by now it works as in Bookmarks in terms of UI
+    #TODO: this should be done through a POST request method
+    collection_sound = get_object_or_404(CollectionSound, id=collectionsound_id)
+    collection = collection_sound.collection
+    collection_sound.delete()
+    return HttpResponseRedirect(reverse("collections", args=[collection.id]))
+
+@login_required
+def get_form_for_collecting_sound(request, sound_id):
+
+    sound = Sound.objects.get(id=sound_id)
+    
+    try:
+        last_collection = \
+            Collection.objects.filter(user=request.user).order_by('-created')[0]
+        # If user has a previous bookmark, use the same category by default (or use none if no category used in last
+        # bookmark)
+    except IndexError:
+        last_collection = None
+    
+    user_collections = Collection.objects.filter(user=request.user).order_by('-created')
+    form = CollectionSoundForm(initial={'collection': last_collection.id if last_collection else CollectionSoundForm.NO_COLLECTION_CHOICE_VALUE},
+                               sound_id=sound.id,
+                               prefix=sound.id,
+                               user_collections=user_collections)
+    
+    collections_already_containing_sound = Collection.objects.filter(user=request.user, collectionsound__sound__id=sound.id).distinct()
+    tvars = {'user': request.user,
+             'sound': sound,
+             'sound_is_moderated_and_processed_ok': sound.moderated_and_processed_ok,
+             'last_collection': last_collection,
+             'collections': user_collections,
+             'form': form,
+             'collections_with_sound': collections_already_containing_sound
+             }
+    
+    return render(request, 'collections/modal_collect_sound.html', tvars)
+
+def delete_collection(request, collection_id):
+    collection = get_object_or_404(Collection, id=collection_id)
+
+    if request.user==collection.user:
```

Comment:
Consider else case, return 404 or reload edit page but with a message saying that user is not allowed